### PR TITLE
Replace `media_url` with `url` in provider scripts

### LINKED ISSUE #1409: Consider using `url` as field name in provider scripts
## Current Situation

Currently in our provider scripts, we use a different field name per media type for the url field -- so, `image_url` and `audio_url`. In the actual database, these all just map to `url`, and so we have to manually convert the field name in each of the associated media stores ([example in ImageStore](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/storage/image.py#L159)).

## Suggested Improvement

Unless there's a reason it must be done this way, I'd like to just use `url` everywhere.

## Benefit

* We would be able to remove the MediaStore logic to convert the field name
* It would be **much easier** to document what fields are needed in provider scripts, and what they map to in the actual database.
* It would be **much easier** for providers that consume media of multiple types to deal with URLs. See [here an example](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py#L184) of Wikimedia needing to set the `url`, then later pop it off the data and rename it depending on the media type, _only for it to just get changed back in the media store_
* It would be **much easier** to define types for `ImageData` and `AudioData`. 

## Additional context

I'm unaware of the reasoning for the current implementation, so it's possible there's a critical reason why these fields are renamed and we should be sure to test thoroughly.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1409 by @stacimc 

## Description

This PR renames `audio_url` and `image_url` in the provider API scripts and `ImageStore`/`AudioStore` classes to `url`, which simplifies the code significantly.

## Testing Instructions

The CI should pass. The tests have been updated to use `url` instead of `image_url`/`audio_url`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
It looks like there's a failing test for the provider data ingester:
```
 =================================== FAILURES ===================================
_____________________________ test_get_batch_data ______________________________
[gw1] linux -- Python 3.10.10 /usr/local/bin/python

    def test_get_batch_data():
        response_json = _get_resource_json("complete_response.json")
        batch = ingester.get_batch_data(response_json)
    
>       assert batch == EXPECTED_BATCH_DATA
E       AssertionError: assert [{'id': 100, ...le 102', ...}] == [{'id': 100, ...534_web.jpg'}]
E         At index 0 diff: {'id': 100, 'image_url': 'https://openaccess-cdn.clevelandart.org/1916.586.a/1916.586.a_web.jpg', 'media_type': 'image', 'title': 'Title 100', 'url': 'https://clevelandart.org/art/1916.586.a'} != {'id': 100, 'media_type': 'image', 'title': 'Title 100', 'url': 'https://openaccess-cdn.clevelandart.org/1916.586.a/1916.586.a_web.jpg'}
E         Use -v to get more diff

tests/dags/providers/provider_api_scripts/test_provider_data_ingester.py:137: AssertionError
=========================== short test summary info ============================
FAILED tests/dags/providers/provider_api_scripts/test_provider_data_ingester.py::test_get_batch_data
```

File: catalog/dags/common/storage/media.py [hunk 1]
```diff
@@ -123,13 +123,13 @@ def clean_media_metadata(self, **media_data) -> dict | None:
         for field in [
             "foreign_identifier",
             "foreign_landing_url",
-            f"{self.media_type}_url",
+            "url",
```
Comment (AetherUnbound) [inline catalog/dags/common/storage/media.py:126]: Whew, this is a nice change to see!

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
I'm also getting the following when I run the SMK DAG locally:

```
[2023-04-25, 00:09:13 UTC] {slack.py:320} INFO - 
*DAG*: `smk_workflow`
*Task*: `ingest_data.pull_image_data`
*Logical Date*: 2023-03-01T00:00:00Z
*Log*: http://localhost:8080/log?execution_date=2023-03-01T00%3A00%3A00%2B00%3A00&task_id=ingest_data.pull_image_data&dag_id=smk_workflow&map_index=-1

*Exception*: ImageStore.add_item() missing 1 required positional argument: 'url'
*Exception Type*: `builtins.TypeError`
```

It looks like this piece was not updated: https://github.com/WordPress/openverse/blob/96fa262619445cb75356c6fa0af7a14242ea3063/catalog/dags/providers/provider_api_scripts/smk.py#L153

I think with a change this large, we'll need to be sure we are able to run each affected DAG successfully locally before we deploy it (I wish there was a better way, maybe some ingestion testing in the future!)

**Comment by obulat** [comment]
> @obulat Is this PR ready for review or it is a work in progress?

I've rebased it onto the license_info PR, and am going to draft it until that PR is merge. I will run the ingestion for all providers locally after that to check that everything works as expected.

**Comment by obulat** [comment]
@AetherUnbound, I ran data ingestion locally, and did not get any `url`-related errors. All but two providers correctly load data. I do get errors in rawpixel (403 response in Airflow, although if I open the link in the browser - it shows results) and iNaturalist (something with local setup in the preingestion tasks).

File: catalog/dags/providers/provider_api_scripts/nypl.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ class NyplDataIngester(ProviderDataIngester):
     # NYPL returns a list of image objects, with the dimension encoded
     # in the URL's query parameter.
     # This list is in order from the largest image to the smallest one.
-    image_url_dimensions = ["g", "v", "q", "w", "r"]
+    url_dimensions = ["g", "v", "q", "w", "r"]
```
Comment (AetherUnbound) [inline catalog/dags/providers/provider_api_scripts/nypl.py:55]: Maybe this would be better as `image_dimensions`?

**Review by AetherUnbound — APPROVED** [review]
This looks great, thanks for testing all the provider scripts! 🙌🏼 I have one thought for a variable rename but it's not blocking.

**Pull request merged**